include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -381,22 +381,25 @@ struct __local_buffer<sycl::buffer<::std::tuple<_T...>, __dim, _AllocT>>
     using type = sycl::buffer<oneapi::dpl::__internal::tuple<_T...>, __dim, _AllocT>;
 };
 
-// __buffer defaulted to sycl::buffer<_T, 1, ...>
+// __buffer_impl defaulted to sycl::buffer<_T, 1, ...>
 template <typename _ExecutionPolicy, typename _T, typename _Container = sycl::buffer<_T, 1>>
-struct __buffer;
+class __buffer_impl;
```

Comment:
```suggestion
template <typename _ExecutionPolicy, typename _T, typename _Container>
class __buffer_impl;
```

That is suggestion based on comment below

Reply:
I think it's not correct suggestion until we delete unused implementation of `class __buffer` :
```C++
template <typename _ExecutionPolicy, typename _T, typename _BValueT, int __dim, typename _AllocT>
class __buffer_impl<_ExecutionPolicy, _T, sycl::buffer<_BValueT, __dim, _AllocT>>
```
Right now we haven't this decision.

Reply:
I think it's not correct suggestion until we delete unused implementation of `class __buffer` :
```C++
template <typename _ExecutionPolicy, typename _T, typename _BValueT, int __dim, typename _AllocT>
class __buffer_impl<_ExecutionPolicy, _T, sycl::buffer<_BValueT, __dim, _AllocT>>
```
Right now we haven't this decision.

Reply:
I think it's not correct suggestion until we delete unused implementation of `class __buffer` :
```C++
template <typename _ExecutionPolicy, typename _T, typename _BValueT, int __dim, typename _AllocT>
class __buffer_impl<_ExecutionPolicy, _T, sycl::buffer<_BValueT, __dim, _AllocT>>
```
Right now we haven't this decision.

Reply:
I think it's not correct suggestion until we delete unused implementation of `class __buffer` :
```C++
template <typename _ExecutionPolicy, typename _T, typename _BValueT, int __dim, typename _AllocT>
class __buffer_impl<_ExecutionPolicy, _T, sycl::buffer<_BValueT, __dim, _AllocT>>
```
Right now we haven't this decision.

Reply:
I think it's not correct suggestion until we delete unused implementation of `class __buffer` :
```C++
template <typename _ExecutionPolicy, typename _T, typename _BValueT, int __dim, typename _AllocT>
class __buffer_impl<_ExecutionPolicy, _T, sycl::buffer<_BValueT, __dim, _AllocT>>
```
Right now we haven't this decision.

Reply:
Fixed

Reply:
Discussed offline, everything should be clear enough now.